marathon_lb.py

```diff
@@ -1529,34 +1530,39 @@ def download_certificates_from_vault(app_map_array, ssl_certs):
     KEY_EXT  = '.key'
     BKP_EXT  = '.bkp'
     O_FORMAT = 'PEM'
+    global previous_app_list
 
     logger.debug("Checking Vault token expiration")
     kms_utils.check_token_needs_renewal(False)    
     #{'/appid1': 'front/backend1'}, {'/folder/appid2': 'front/backend2'}
     logger.debug("Download certificates for appid (if not exists) from Vault")
     currentListAppidCert = list()
     for app in app_map_array:
-        
-
-        # TODO: When Vault will adapt the structure to allow store the tenant concept
-        # we will need to read the full path, plain it, or whatever.
-        #appid = list(app.keys())[0].replace('/','')
-        appid = list(app.keys())[0].split('/')[-1]
 
+        # appid should follow multitenant convention: replace "/" with "." and reverse the order
+        # So if app is /test/nginx, appid should be nginx.test
+        # Old appid is next comment 
+        #appid = list(app.keys())[0].split('/')[-1]        
+        appid = '.'.join(list(app.keys())[0].split('/')[::-1][:-1])
```

Comment:
With this expression we're retrieving `nginx` from `/test/nginx`. Is this what we want?

If we need to retrieve `nginx.test` from `/text/nginx`, the expression should be:

`appid = '.'.join(list(app.keys())[0].split('/')[::-1])`